internal/consul/namespaces.go

```diff
@@ -29,8 +41,17 @@ func EnsureNamespaceExists(client Client, ns string) (bool, error) {
 		return false, nil
 	}
 
-	// If not, create it.
-	var aclConfig capi.NamespaceACLConfig
+	// If the namespace does not, create it with default cross-namespace-policy.
+	crossNamespacePolicy, err := getOrCreateCrossNamespacePolicy(client, partitionInfo)
```

Comment:
yeah I don't think we want to patch existing namespaces for that reason you mentioned: they're not managed by us and might give broader permissions than the operator wants (it could be a bit of a footgun, but if the operator is manually creating namespaces I'd expect them to also be managing the permissions)